bookshelf/bookshelf.d.ts

```diff
@@ -177,8 +191,12 @@ declare module 'bookshelf' {
 			keys() : string[];
 			last() : T;
 			lastIndexOf(value : any, fromIndex? : number) : number;
-			map(predicate? : Lodash.ListIterator<T, boolean>|Lodash.DictionaryIterator<T, boolean>|string, thisArg? : any) : T[];
+
+			// See https://github.com/Microsoft/TypeScript/blob/v1.8.10/lib/lib.core.es7.d.ts#L1122
+			map<U>(predicate? : Lodash.ListIterator<T, U>|string, thisArg? : any) : U[];
```

Comment:
Very good question.

See the test here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/9297/files#diff-8a3091671ada9e45452b597b82364fddR214, line:

```JavaScript
const booksOutput = books.map(book => new Author2Output(book.toJSON()));
```

`book` should be of type `Book` (and `booksOutput` of type `AuthorOutput[]`)
- works with `map<U>(predicate? : Lodash.ListIterator<T, U>|string, thisArg? : any) : U[];`
- works with `map<U>(predicate? : Lodash.DictionaryIterator<T, U>|string, thisArg? : any) : U[];`
- does not work (`book` is of type `any`) with `map<U>(predicate? : Lodash.ListIterator<T, U>|Lodash.DictionaryIterator<T, boolean>|string, thisArg? : any) : U[];`

TypeScript is lost with `Lodash.ListIterator<T, U>|Lodash.DictionaryIterator<T, U>`, I don't know why.

<br>

This:
```JavaScript
map<U>(predicate? : Lodash.ListIterator<T, U>|string, thisArg? : any) : U[];
map<U>(predicate? : Lodash.DictionaryIterator<T, U>|string, thisArg? : any) : U[];
```
works => the best solution?

I don't know the differences between `ListIterator` and `DictionaryIterator`, not a lodash expert :)